aws-android-sdk-polly/src/test/java/com/amazonaws/services/polly/AmazonPollyPresigningClientTest.java

```diff
@@ -216,4 +213,39 @@ private void assertBasicUriValues(URI uri) {
         Assert.assertEquals(PORT, uri.getPort());
         Assert.assertEquals(RESOURCE_PATH, uri.getPath());
     }
+
+    private static final class QueryParameters {
+        private QueryParameters() {}
+
+        static List<Pair> parse(URI uri) {
+            List<Pair> list = new ArrayList<>();
+            String query = uri.getQuery();
+            if (query == null) {
+                return Collections.emptyList();
+            }
+            for (String pair : query.split("&")) {
+                if (pair != null && pair.length() > 0) {
+                    String[] parts = pair.split("=");
+                    list.add(new Pair(parts[0], parts[1]));
```

Comment:
I think you need to UTF-8 decode the `key` and `value` here?   

Also, should we be a little more defensive and make sure parts.length == 2?

Reply:
Resolution: I went back to using the `UrlQuerySanitizer`, so this method has been removed.